web/packages/teleport/src/Discover/Shared/Finished/Finished.tsx

```diff
@@ -55,8 +61,17 @@ export function Finished(props: AgentStepProps & Message) {
   if (props.resourceText) {
     resourceText = props.resourceText;
   }
-  if (props.redirect) {
-    redirect = props.redirect;
+  if (props.primaryButtonText) {
+    primaryButtonText = props.primaryButtonText;
+  }
+  if (props.primaryButtonAction) {
+    primaryButtonAction = props.primaryButtonAction;
+  }
+  if (props.secondaryButtonText) {
+    secondaryButtonText = props.secondaryButtonText;
+  }
+  if (props.secondaryButtonAction) {
+    secondaryButtonAction = props.secondaryButtonAction;
   }
```

Comment:
I would remove setting these 4 values with `let` and then conditionally setting them, and instead, use the "default value pattern" instead

```ts
export function Finished({
  primaryButtonAction = () => history.push(cfg.routes.root, true),
  primaryButtonText = 'Browse Existing Resources',
  secondaryButtonAction = () => history.reload(),
  secondaryButtonText = 'Add Another Resource',
  ...props
}: AgentStepProps & Message) {
```

i know we did the "overwriting pattern" for other props in this file but, we can keep down the changes in the PR using this pattern and fix the old stuff at another time.
```suggestion
```